src/org/zaproxy/zap/extension/ascanrulesAlpha/TestUserAgent.java

```diff
@@ -0,0 +1,145 @@
+
+    private static Logger log = Logger.getLogger(TestUserAgent.class);
+
+    private static final int PLUGIN_ID = 30897;
+
+    private static final String InternetExplorer8 = "Mozilla/4.0 (compatible; MSIE 8.0; Windows NT 6.1)";
+    private static final String InternetExplorer7 = "Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 6.0)";
+    private static final String InternetExplorer6 = "Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.1)";
+    private static final String GoogleBot2_1 = "Mozilla/5.0 (compatible; Googlebot/2.1; +http://www.google.com/bot.html)";
+    private static final String MsnBot1_1 = "msnbot/1.1 (+http://search.msn.com/msnbot.htm)";
+    private static final String YahooSlurp = "Mozilla/5.0 (compatible; Yahoo! Slurp; http://help.yahoo.com/help/us/ysearch/slurp)";
+    private static final String iPhone3 = "Mozilla/5.0 (iPhone; U; CPU iPhone OS 3_0 like Mac OS X; en-us) AppleWebKit/528.18 (KHTML, like Gecko) Version/4.0 Mobile/7A341 Safari/528.16";
+
+    private static final String[] USER_AGENTS = {
+            InternetExplorer8,
+            InternetExplorer7,
+            InternetExplorer6,
+            GoogleBot2_1,
+            MsnBot1_1,
+            YahooSlurp,
+            iPhone3
+    };
+
+    @Override
+    public int getId() {
+        return PLUGIN_ID;
+    }
+
+    @Override
+    public String getName() {
+        return Constant.messages.getString("ascanalpha.testuseragent.name");
+    }
+
+    @Override
+    public String[] getDependency() {
+        return new String[0];
+    }
+
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString("ascanalpha.testuseragent.desc");
+    }
+
+    @Override
+    public int getCategory() {
+        return Category.INFO_GATHER;
+    }
+
+    @Override
+    public String getSolution() {
+        return "";
+    }
+
+    @Override
+    public String getReference() {
+        return Constant.messages.getString("ascanalpha.testuseragent.refs");
+    }
+
+    @Override
+    public void init() {
+    }
+
+    @Override
+    public void scan() {
+        for(String userAgent: USER_AGENTS)
+        {
+            attack(userAgent);
+        }
+    }
+
+    private void attack(String userAgent){
+        HttpMessage newMsg = sendUserAgent(userAgent);
+        if(newMsg != null && isResponseDifferentFromOriginal(newMsg)){
+            createAlert(newMsg, userAgent);
+        }
+    }
+
+    private HttpMessage sendUserAgent(String userAgent){
+        try {
+            HttpMessage newMsg = getNewMsg();
+            HttpRequestHeader header = newMsg.getRequestHeader();
+            header.setHeader("User-Agent", userAgent);
+            sendAndReceive(newMsg);
+            return newMsg;
+        } catch (Exception e) {
+            log.error(e.getMessage(), e);
+        }
+        return null;
+    }
+
+    private boolean isResponseDifferentFromOriginal(HttpMessage newMsg){
+        HttpMessage originalMsg = this.getBaseMsg();
```

Comment:
I think a comment would not be noticed. I created the Issue https://github.com/zaproxy/zaproxy/issues/1964